### PR TITLE
Bug/fix error messag display for business rules

### PR DESCRIPTION
**Pull request by rkrusselenq**
## Description

There was a bug caught when testing the error message display for date rules.  There was some form fields automatically being reset, so i moved that logic into a change handler instead of in useEffect.
## Tickets

* [Jira Ticket](https://cdc-nbs.atlassian.net/jira/software/c/projects/CNFT2/boards/99?assignee=712020%3A591f5469-427d-49e9-87dc-8b4df72a02b4&selectedIssue=CNFT2-2120)

## Checklist before requesting a review
- [x] PR focuses on a single story
- [x] Code has been fully tested to meet acceptance criteria
- [x] PR is reasonably small and reviewable (Generally less than 10 files and 500 changed lines)
- [x] All new functions/classes/components reasonably small
- [x] Functions/classes/components focused on one responsibility
- [x] Code easy to understand and modify (clarity over concise/clever)
- [x] PRs containing TypeScript follow the [Do's and Don'ts](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html)
- [x] PR does not contain hardcoded values (Uses constants)
- [x] All code is covered by unit or feature tests

### CODE REVIEW

File: apps/modernization-ui/src/apps/page-builder/pages/BusinessRulesLibrary/Add/AddBusinessRule.tsx [hunk 1]
```diff
@@ -24,13 +24,16 @@ const AddBusinessRule = () => {
...
+    const [loading, setIsLoading] = useState(false);
 
     useEffect(() => {
         if (ruleId) {
+            setIsLoading(true);
             PageRuleControllerService.viewRuleResponseUsingGet({
                 authorization: authorization(),
                 ruleId: Number(ruleId)
             }).then((resp: Rule) => {
+                console.log('response', resp);
```
Comment (alaapbharadwaj) [inline apps/modernization-ui/src/apps/page-builder/pages/BusinessRulesLibrary/Add/AddBusinessRule.tsx:36]: Issue: console

File: apps/modernization-ui/src/apps/page-builder/pages/BusinessRulesLibrary/helpers/mapLogicForDateCompare.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import { Rule } from 'apps/page-builder/generated';
 
 export const mapLogicForDateCompare = (logic: Rule.comparator) => {
+    console.log('logic', logic);
```
Comment (alaapbharadwaj) [inline apps/modernization-ui/src/apps/page-builder/pages/BusinessRulesLibrary/helpers/mapLogicForDateCompare.ts:4]: Issue: another console

**Review by alaapbharadwaj — CHANGES_REQUESTED** [review]

File: apps/modernization-ui/src/apps/page-builder/pages/BusinessRulesLibrary/helpers/mapLogicForDateCompare.ts (hunk 2 above)
Comment (rkrusselenq) [reply apps/modernization-ui/src/apps/page-builder/pages/BusinessRulesLibrary/helpers/mapLogicForDateCompare.ts]: fixed

File: apps/modernization-ui/src/apps/page-builder/pages/BusinessRulesLibrary/Add/AddBusinessRule.tsx (hunk 1 above)
Comment (rkrusselenq) [reply apps/modernization-ui/src/apps/page-builder/pages/BusinessRulesLibrary/Add/AddBusinessRule.tsx]: fixed

**Review by mpeels — APPROVED** [review]

**Review by alaapbharadwaj — APPROVED** [review]

**Pull request merged**